File: Perl言語/基礎知識用の勉強/5関数勉強/関数.pl
use v5.24;

my @asakuno = ("朝来野", "智博", );

sub asakuno {
	say "関数：@_";
}

# 以下2つの関数は、関数2つ目の実装前に実行した結果。
&asakuno();	# 関数：
&asakuno("関数呼び出し");	# 関数：関数呼び出し
#	2つ目の関数を実装後に実行した場合、2つ目の関数を実行する。
#	この理由は、ファイル後方にある関数が、ファイル先頭にある関数を上書きするため。
#	そのため、同名関数は作るべからず。

say "-" x 30;

sub asakuno {
	"関数2つ目：@_";
}
say &asakuno();	# 関数2つ目：
my $ret = &asakuno("関数呼び出し");	# 関数2つ目：関数呼び出し
say $ret;

# 上記の関数は、最後の処理が戻り値になるため、その戻り値を呼び出し側で出力(say)している。


say "以上。"
# vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=perl noexpandtab:

File: Go言語/基礎知識用の勉強/5関数勉強/実行済み/関数.go
package main

import "fmt"

func asakuno() {
	fmt.Println("asakuno関数内からの出力(引数なし-戻り値なし)")
}

func tomohiro1() {
	fmt.Println("asakuno関数内からの出力(引数なし-戻り値なし)")
}

func tomohiro2( arg [10]int ) {
	fmt.Printf("tomohiro2関数内からの出力(引数%d-戻り値なし)\n", arg[0])
}

func tomohiro3() (ret int) {
	ret = 20211001
	fmt.Println("tomohiro3関数内からの出力(引数なし-戻り値あり)")
	return
}

func tomohiro4( arg string ) int {
	fmt.Printf("tomohiro4関数内からの出力(引数%s-戻り値あり)\n", arg)
	return 9784798063997
}

func main() {
	var asakuno [10]int;
	asakuno[0] = 20210929;
	fmt.Println("-----------------------");

	//asakuno()	// エラー
	// # command-line-arguments
	// cannot call non-function asakuno (type [10]int)
	// 以下、関数名と変数名が一致しない場合の呼び出し結果。
	//	asakuno関数内からの出力(引数なし-戻り値なし)

	tomohiro1()
	// asakuno関数内からの出力(引数なし-戻り値なし)
	fmt.Println("-----------------------");

	tomohiro2( asakuno )
	// tomohiro2関数内からの出力(引数20210929-戻り値なし)
	fmt.Println("-----------------------");

	retint := tomohiro3( )
	// tomohiro3関数内からの出力(引数なし-戻り値あり)
	fmt.Println("tomohiro3からの戻り値：", retint)
	// tomohiro3からの戻り値： 20211001
	fmt.Println("-----------------------");

	retstr := tomohiro4( "朝来野智博" )
	// tomohiro4関数内からの出力(引数朝来野智博-戻り値あり)
	fmt.Println("tomohiro4からの戻り値：", retstr)
	// tomohiro4からの戻り値： 9784798063997
	fmt.Println("-----------------------");

}

/* vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=go noexpandtab: */
